### ISSUE TITLE
reverse bit and byte order of DMR trellis decoded data

### ISSUE THREAD

**Issue by iddq** [g4klx/MMDVMHost#744]
Hi Jonathan,

I assume the bits and bytes from trellis decoder are in reverse order which may wrong... This is true for data from both the MMDVM and the network.

to reproduce the issue I sent text in motorola format which use data rate 3/4 and used the code below:

https://github.com/iddq/MMDVMHost/commit/6a9467c395900a6115c22d44c9ad34cd2a12de90

I've tried to compare it with other implementation but it is beyond my knowledge.

https://github.com/pd0mz/go-dmr/blob/ecea90fd7937aa1b145f304e28a3870790b161ff/terminal/terminal.go#L506

**Comment by iddq** [comment]
cc: @g4klx

**Comment by lwvmobile** [comment]
I was actually wondering about the trellis decoding myself. I was attempting to implement the decoding function into a project as well, and the values I was getting out weren't returning as expected. I attempted to reverse the order of the dibits, reverse all the bits, reverse the payload, reversed the bits of the payload, and still get non expected values on the 3/4 rate data. I put a trace on each individual step, and when it gets to the checkcode function to make tribits, I get tribit values returned that are much greater than 3 bits. 

I've attached a text file with a small burst of 3/4 rate data (including the initial dibits) which other decoders tell me is LRRP data, and includes dumps of the data at each step of the decode process.

[trellis.zip](https://github.com/g4klx/MMDVMHost/files/8349676/trellis.zip)
